### PR TITLE
Implement `Add` operator for Onnx

### PR DESCRIPTION
**Pull request by Luni-4**
### Checklist

- [x] Confirm that `run-checks.sh` has been executed.

### Related Issues/PRs

_Provide links to relevant issues and dependent PRs._

### Changes

This PR implements the `Add` operator for onnx

### Testing

Run `cargo build` and `cargo test`

### CODE REVIEW

File: burn-import/src/burn/node/add.rs [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+use burn::record::PrecisionSettings;
+use proc_macro2::TokenStream;
+use quote::quote;
+
+#[derive(Debug, Clone, new)]
+pub struct AddNode {
+    pub lhs: TensorType,
+    pub rhs: TensorType,
+    pub output: TensorType,
+}
```
Comment (nathanielsimard) [inline burn-import/src/burn/node/add.rs:12]: This will be very similar for all unary operations. I think it would be interesting to have a way to avoid code duplication with a trait that only defines the forward method. It may be done in another PR.

**Comment by Luni-4** [comment]
@nathanielsimard 

My concern is that my code does not consider adding a scalar to a tensor at all. I do not know how to discriminate that case

File: burn-import/src/burn/node/add.rs (hunk 1 above)
Comment (Luni-4) [reply burn-import/src/burn/node/add.rs:12]: Yes, that could be an idea. I agree

**Review by antimora — APPROVED** [review]
Looks good.
